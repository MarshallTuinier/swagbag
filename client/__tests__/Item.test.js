import ItemComponent from "../Components/Item";
import { shallow } from "enzyme";

const fakeItem = {
  id: "ABC123",
  title: "Sweet Item",
  price: 5000,
  description: "Reaaallllyyy sweet item.",
  image: "sickThing.jpg",
  largeImage: "largeSickThing.jpg"
};

describe("<Item />", () => {
  const wrapper = shallow(<ItemComponent item={fakeItem} />);
  it("renders and displays price/title", () => {
    const PriceTag = wrapper.find("Item__PriceTag");
    expect(PriceTag.children().text()).toBe("$50");
    expect(wrapper.find("Item__Title a").text()).toBe(fakeItem.title);
  });
  it("renders the image properly", () => {
    const img = wrapper.find("img");
    expect(img.props().src).toBe(fakeItem.image);
    expect(img.props().alt).toBe(fakeItem.title);
  });

  it("renders buttons properly", () => {
    const buttonList = wrapper.find(".buttonList");
    expect(buttonList.children()).toHaveLength(3);
    expect(buttonList.find("Link")).toHaveLength(1);
    expect(buttonList.find("AddToCart").exists()).toBe(true);
    expect(buttonList.find("DeleteItem").exists()).toBe(true);
  });
});
