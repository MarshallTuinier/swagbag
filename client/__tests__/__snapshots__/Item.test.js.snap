// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matchest the snapshot 1`] = `
<Item__StyledItem>
  <img
    alt="Sweet Item"
    src="sickThing.jpg"
  />
  <Item__Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Sweet Item
      </a>
    </Link>
  </Item__Title>
  <Item__PriceTag>
    $50
  </Item__PriceTag>
  <p>
    Reaaallllyyy sweet item.
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit 📝
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    >
      Delete This Item 🚫
    </DeleteItem>
  </div>
</Item__StyledItem>
`;
